plugins/nf-wave/src/main/io/seqera/wave/plugin/WaveAssets.groovy

```diff
@@ -81,6 +83,8 @@ class WaveAssets {
         allMeta.add( this.spackFile?.text )
         allMeta.add( this.projectResources?.fingerprint() )
         allMeta.add( this.containerPlatform )
+        allMeta.add( this.spackFile ? this.spackTarget : null )
```

Comment:
Look at the `spack_arch` in the Dockerfile template (same holds for the Singularityfile template):  https://github.com/seqeralabs/wave/blob/master/src/main/resources/io/seqera/wave/spack/spack-builder-dockerfile.txt#L27

This has to take the value that comes from the Wave Client here in Nextflow